src/main/scala/sbtprotoc/ProtocPlugin.scala

```diff
@@ -217,13 +217,18 @@ object ProtocPlugin extends AutoPlugin with Compat {
   private[this] def unpack(
       deps: Seq[File],
       extractTarget: File,
-      log: Logger
+      streams: TaskStreams
   ): Seq[(File, Seq[File])] = {
-    IO.createDirectory(extractTarget)
+    val cachedExtractDeps = FileFunction.cached(streams.cacheDirectory / "protobuf_external") { deps =>
+      IO.createDirectory(extractTarget)
+      deps.flatMap { dep =>
+        val set = IO.unzip(dep, extractTarget, "*.proto")
+        if (set.nonEmpty) streams.log.debug("Extracted " + set.mkString("\n * ", "\n * ", ""))
+        set
+      }
+    }
     deps.map { dep =>
-      val seq = IO.unzip(dep, extractTarget, "*.proto").toSeq
-      if (seq.nonEmpty) log.debug("Extracted " + seq.mkString("\n * ", "\n * ", ""))
-      dep -> seq
+      dep -> cachedExtractDeps(Set(dep)).toSeq
```

Comment:
given the signature of  the closure wrapped by `FileFunction.cached` and because we need to keep the mapping of extracted protos per dependency, the cache has to be per dependency